Subject to https://github.com/PaddlePaddle/Paddle/pull/46770 impact, the accuracy of mkldnn inference for two models(deeplabv3  and det_mv3_db) decreases

### bug描述 Describe the Bug

**Error Description：** After the merge of https://github.com/PaddlePaddle/Paddle/pull/46770 the MKL-DNN inference accuracy of the deeplabv3 and det_mv3_db models has decreased.

The max_diff comparison:

- test_det_mv3_db_mkldnn.py 
   before：3.7431717e-05 
   after: 0.99988633

- test_deeplabv3_mkldnn.py 
   before：5.841255e-06 
   after: 0.90

**1. Operating environment**
PaddlePaddle version：develop
OS version：CentOS
Python version：3.8

**2. The detailed error information: (The comparison strategy is to exit when the first one is greater than the diff threshold)**
```test_det_mv3_db_mkldnn.py F                                                                   [100%]

============================================= FAILURES ==============================================

test_det_mv3_db_mkldnn.py:48:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <test_mkldnn_helper.TestModelInferenceMKLDNN object at 0x7fd3cca15730>
result = array([2.9321874e-22, 1.5445235e-22, 1.8817962e-19, ..., 1.4314117e-16,
       4.2289070e-16, 3.1968471e-18], dtype=float32)
expect = array([2.9322545e-22, 1.5445590e-22, 1.8818681e-19, ..., 1.4314335e-16,
       4.2289555e-16, 3.1968593e-18], dtype=float32)
delta = 0.0001

    def check_data(self, result, expect, delta):
        """
        check result
        Args:
            result(list): list of result data
            expect(list): list of expect data
            delta(float): e.g. 0.001
        Returns:
            None
        """
        logger.info("current comparison delta is : {0}".format(delta))
        assert len(result) == pytest.approx(len(expect)), "output length not equal"
        for i in range(0, len(expect)):
>           assert result[i] == pytest.approx(expect[i], rel=None, abs=delta), "output data not equal"
E           AssertionError: output data not equal
E           assert 0.0068414505 == 3.55381445160...e-07 ± 1.0e-04
E             comparison failed
E             Obtained: 0.006841450463980436
E             Expected: 3.5538144516067405e-07 ± 1.0e-04

test_mkldnn_helper.py:60: AssertionError
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

test_deeplabv3_mkldnn.py:48:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <test_mkldnn_helper.TestModelInferenceMKLDNN object at 0x7f13018327f0>
result = array([9.283574e-03, 9.283574e-03, 9.283574e-03, ..., 7.704613e-05,
       7.704613e-05, 7.704613e-05], dtype=float32)
expect = array([9.283591e-03, 8.969743e-03, 8.659990e-03, ..., 8.441107e-05,
       8.067604e-05, 7.704619e-05], dtype=float32)
delta = 0.0001

    def check_data(self, result, expect, delta):
        """
        check result
        Args:
            result(list): list of result data
            expect(list): list of expect data
            delta(float): e.g. 0.001
        Returns:
            None
        """
        logger.info("current comparison delta is : {0}".format(delta))
        assert len(result) == pytest.approx(len(expect)), "output length not equal"
        for i in range(0, len(expect)):
>           assert result[i] == pytest.approx(expect[i], rel=None, abs=delta), "output data not equal"
E           AssertionError: output data not equal
E           assert 0.009283574 == 0.008969742804765701 ± 1.0e-04
E             comparison failed
E             Obtained: 0.009283574298024178
E             Expected: 0.008969742804765701 ± 1.0e-04

```
**3. The problem recurrence steps：**
```
git clone https://github.com/PaddlePaddle/continuous_integration.git --depth=1

cd ./continuous_integration/inference/inference_api_test/python_api_test/

project_path=`pwd`
export project_path
cd ${project_path}

# download Data

mkdir -p ./Data
cd ./Data

# download models
wget --no-proxy -q https://sys-p0.bj.bcebos.com/inference/python-model-infer.tgz --no-check-certificate
tar -xvf python-model-infer.tgz

# download ocr models
wget --no-proxy -q https://sys-p0.bj.bcebos.com/inference/python-ocr-infer.tgz --no-check-certificate
tar -xvf python-ocr-infer.tgz

cd -

# requirements
python -m pip install --upgrade pip -i https://mirror.baidu.com/pypi/simple
python -m pip install -r requirements.txt -i https://mirror.baidu.com/pypi/simple

# download paddlepaddle_whl （Optional）

# latest(failed)
wget -q https://paddle-qa.bj.bcebos.com/paddle-pipeline/Develop-GpuAll-LinuxCentos-Gcc82-Cuda112-Trtoff-Py38-Compile/latest/paddlepaddle_gpu-0.0.0-cp38-cp38-linux_x86_64.whl

## error_pr(failed)
#wget -q https://paddle-qa.bj.bcebos.com/paddle-pipeline/Develop-GpuAll-LinuxCentos-Gcc82-Cuda112-Trtoff-Py38-Compile/8e6315e4a18c5a5cf19d8949301d384647682598/paddlepaddle_gpu-0.0.0-cp38-cp38-linux_x86_64.whl

## the previous commit of error_pr(passed)
#wget -q https://paddle-qa.bj.bcebos.com/paddle-pipeline/Develop-GpuAll-LinuxCentos-Gcc82-Cuda112-Trtoff-Py38-Compile/e23dfed9ea00f69acd4ba90583d5e355cdaa3b59/paddlepaddle_gpu-0.0.0-cp38-cp38-linux_x86_64.whl

# install paddlepaddle_whl
python -m pip install paddlepaddle_gpu-0.0.0-cp38-cp38-linux_x86_64.whl


# run case
cd ./tests/mkldnn

python -m pytest test_det_mv3_db_mkldnn.py
python -m pytest test_deeplabv3_mkldnn.py
```

### 其他补充信息 Additional Supplementary Information
It is also positioned that the reason of this problem mainly as the impact of an interpolate_ mkldnn_pass. There's a solution that you can add 'predictor_config.delete_pass('interpolate_mkldnn_pass') ' below continuous_integration/inference/inference_api_test/python_api_test/src/test_case.py line 102 https://github.com/PaddlePaddle/continuous_integration/blob/develop/inference/inference_api_test/python_api_test/src/test_case.py#:~:text=set_cpu_math_library_num_threads

```
elif config_type == 'mkldnn':
     predictor_config.disable_gpu()
     predictor_config.enable_mkldnn()
     predictor_config.set_cpu_math_library_num_threads(4)
     predictor_config.delete_pass('interpolate_mkldnn_pass')
```

--- a/paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass.cc
+++ b/paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass.cc
@@ -69,8 +69,24 @@ inline bool FoundPhiOneDNNKernelWithCorrectDataType(
 
 bool MKLDNNPlacementPass::IsSupport(const Node* op) const {
   if (FoundOneDNNKernelWithCorrectDataType(op) ||
-      FoundPhiOneDNNKernelWithCorrectDataType(op))
-    return true;
+      FoundPhiOneDNNKernelWithCorrectDataType(op)) {
+    // For interpolate ops, there's a little difference between Paddle and
+    // DNNL.
+    // If run DNNL interpolate ops, manual set AnalysisConfig and apply
+    // the corresponding pass.
+    const std::vector<std::string> not_default_op_types = {"bilinear_interp",
+                                                           "nearest_interp",
+                                                           "trilinear_interp",
+                                                           "bicubic_interp",
+                                                           "linear_interp",
+                                                           "bilinear_interp_v2",
+                                                           "linear_interp_v2"};
+    bool is_interpolate_op =
+        std::find(not_default_op_types.begin(),
+                  not_default_op_types.end(),
+                  op->Op()->Type()) != not_default_op_types.end();
+    return !is_interpolate_op;
+  }
   return false;
 }
 

--- a/paddle/fluid/inference/api/paddle_pass_builder.cc
+++ b/paddle/fluid/inference/api/paddle_pass_builder.cc
@@ -363,7 +363,6 @@ void CpuPassStrategy::EnableMKLDNN() {
              "conv_transpose_eltwiseadd_bn_fuse_pass",  //
              "conv_bias_mkldnn_fuse_pass",              //
              "conv_transpose_bias_mkldnn_fuse_pass",
-             "interpolate_mkldnn_pass",
              // TODO(baoachun): Need to support 5-dimensional input.
              // "conv3d_bias_mkldnn_fuse_pass",  //
              "conv_elementwise_add_mkldnn_fuse_pass",

